src/octoprint/plugins/softwareupdate/__init__.py

```diff
@@ -704,6 +704,7 @@ def get_settings_defaults(self):
             "updatelog_cutoff": 30 * 24 * 60,  # 30 days
             "credentials": {
                 "github": None,
+                "github_set": False,
```

Comment:
This should not be part of the settings defaults as it is inherently tied to the value of `credentials.github`, but rather be set on the API response via `on_settings_load`. That would then also mean the change to `get_settings_restricted_paths` can be rolled back again, and would simplify the logic in `on_settings_load`.